### PR TITLE
Testing for the kick handler

### LINKED ISSUE #308: Write unit tests for the kick handler
# Summary

Unit tests do not currently exist for the kick handler. 

# Background

**Is your improvement related to a problem? Please describe**:

After writing the kick handler, it is unclear if we have handled all edge cases. We should have unit tests to thoroughly test the kick handler functionality.

**Describe the solution you'd like**:

Full set of unit tests for the kick handler to catch as many edge cases as we can.

**Describe alternatives you've considered**:

No alternatives.

# Details

The owner of this issue needs to think of what the kick handler should support, and not support. Then, build unit tests around the desired and undesired functionality to make sure error handling exists in the appropriate locations.

# Outcome

Unit testing encompasses the kick handler. Allows us to feel safe knowing the kick handler has been thoroughly tested.

### PR DESCRIPTION
**Pull request by 10eMyrT**
This PR introduces testing for the kick handler. Thanks to @Zedjones for his interface and the join/part handlers testing.

Closes #308

### CODE REVIEW

**Comment by 10eMyrT** [comment]
@Tjzabel Thabks for the review. I will wait for @Zedjones's approval to merge.

**Comment by jwflory** [comment]
Just a note, this needs to be rebased on `master`.

File: internal/handlers/irc/handlers_test.go [hunk 1]
```diff
@@ -205,3 +205,111 @@ func TestQuitHandler_Off(t *testing.T) {
...
+	mockLogger.
+		EXPECT().
+		LogDebug(gomock.Eq("kickHandler triggered"))
+	mockClient.
+		EXPECT().
+		TgSettings().
+		Return(&tgSettings)
+	mockClient.
+		EXPECT().
+		SendToTg(gomock.Eq("* TEST_NAME kicked TEST_KICKEDNAME from TEST_GROUP: TEST_KICKEDNAME"))
```
Comment (Zedjones) [inline internal/handlers/irc/handlers_test.go:303]: I am curious if this is the behavior we want. It seems a bit redundant to show the username again, but also not related to this review. Opinions? @jwflory @Tjzabel

Comment (jwflory) [reply internal/handlers/irc/handlers_test.go:303]: @Zedjones If possible, I would prefer a kick reason to be substituted there. It's worth noting that a kick reason may not always be specified. Should we open a new issue for this?

Comment (10eMyrT) [reply internal/handlers/irc/handlers_test.go:303]: I updated the code to display "Reason Undefined" as reason if none is provided to avoid redundancy.
